src/main/sensors/gyro.c

```diff
@@ -681,7 +682,12 @@ STATIC_UNIT_TESTED void performGyroCalibration(gyroSensor_t *gyroSensor, uint8_t
                 gyroSetCalibrationCycles(gyroSensor);
                 return;
             }
-            gyroSensor->gyroDev.gyroZero[axis] = (gyroSensor->calibration.sum[axis] + (gyroCalculateCalibratingCycles() / 2)) / gyroCalculateCalibratingCycles();
+
+            if (axis != 2) {
```

Comment:
'axis != Z'

Also it may be beter to write is this way:
```
gyroSensor->gyroDev.gyroZero[axis] = gyroSensor->calibration.sum[axis] / gyroCalculateCalibratingCycles();
if (axis == Z) {
  gyroSensor->gyroDev.gyroZero[axis] -= ((float)gyroConfig()->gyro_offset_yaw / 100));
}
```

And another problem is board alignment - if gyro Z isn't aligned with yaw, this will compensate wrong axis ...

Reply:
@ledvinap 
hmm ...
In alignSensors i see only  Z and -Z (normal and flipped)
Can you point me in the right direction ?

Reply:
Yes, I didn't check it. Only 'reasonable' alignments are enumerated

`alignBoard` may still cause problem.

But for now, it is probably OK. Deeper redesign would eventually fix it.
It is good idea to document it, so that someone with exotic configuration is not surprised ..